Automattic/wp-calypso
# Anchor: Gutenboarding Launch works immediately without domain or plan selection

#### Changes proposed in this Pull Request

* After creating a site with the Anchor Gutenboarding process, the 'Launch' button displayed in the editor will launch the site without asking the user to select a domain or plan.

#### Note: Gutenboarding Launch Methods
* There are currently has 3 different ways to handle site launch. "Step By Step", "Focused Launch", and "Calypso Launch."
* The code that determines which launch style is used is [here](https://github.com/Automattic/wp-calypso/blob/5b0a61b0fc53680a98c2bdd7cb205fc245f7e496/apps/editing-toolkit/editing-toolkit-plugin/editor-gutenboarding-launch/index.ts#L120). 
* "Focused Launch" looks to be a newer experiment that is only turned on with a feature flag, and there are references to future A/B tests in the comments.
* Currently, free sites made with gutenboarding are sent to "Step By Step" launch. This covers all anchor gutenboarding sites.
  * The code in this PR only covers "Step by Step" launch.
  * I added an extra conditional to always send Anchor Gutenboarding sites to "Step by Step" launch: c7608994698fe6. While this condition is currently superfluous, my goal is to make Anchor Gutenboarding more robust against future changes by other teams.  However, I'm open to removing it if it's considered unneeded complexity.

#### Testing instructions

* Apply patch D56138-code.
* Need to build and sync `apps/editing-toolkit` from this branch to sandbox
  * In a terminal tab:
  * `cd ./apps/editing-toolkit`
  * `yarn dev --sync`
  * Changes should be in `wp-content/plugins/full-site-editing-plugin/dev/editor-gutenboarding-launch/dist/editor-gutenboarding-launch.js` & also `launch-button.js`.
    * If changes are there and it's not working, edit a post, click the vertical three dots in the top right, and at the bottom of the sidebar, verify the version of the editing toolkit being loaded is `dev`
* Need to create new site using anchor gutenboarding
  * Run calypso locally
  * Sandbox public-api
  * Visit http://calypso.localhost:3000/new?&anchor_podcast=22b6608&anchor_episode=e324a06c-3148-43a4-85d8-34c0d8222138&spotify_url=https%3A%2F%2Fopen.spotify.com%2Fshow%2F6HTZdaDHjqXKDE4acYffoD%3Fsi%3DEVfDYETjQCu7pasVG5D73Q and create a new site. You can either use incognito and make a new account or log into an existing account.
  * Go through all anchor gutenboarding steps.
  * You should end on an editor page with the podcast player embedded.  Keep that tab/URL.
* Need to sandbox that exact site you just created
  * Check the URL to see the site you created, and add that to your /etc/hosts to point to your sandbox.  For example, it might look like `bravenotperfectwithreshmasaujani340284555.wp.com`
  * Clear DNS cache in browser OR restart browser OR wait 3-5 minutes
    * Firefox: Visit `about:networking#dns` -> Clear Cache
    * You might have to close, reopen and relogin with any incognito windows? I think they keep their own DNS cache that can't be cleared for some reason.
  * Visiting `/wp-admin` on it should show sandboxed banner
  * F5 your editor page. Very important: If you press the three vertical dots in the top right, the bottom of the sidebar should read `Editing Toolkit: dev`. If it doesn't, you won't see the changes here.
    * I have struggled to get this working even though I have done everything "right" in setup.  You can try restarting the browser, or restarting autoproxxy, or as a hacky workaround, copy the contents of `wp-content/plugins/full-site-editing-plugin/dev/` in your sandbox over `wp-content/plugins/full-site-editing-plugin/49211111111/` (change number to the one being displayed).
* Press the Launch button in the top right
  * Your site should launch immediately without being asked for domain or plan upsells
* Test the negative: (Does this PR only affect anchor?) 
  * Create a new gutenboarding site that isn't anchor (Use only http://calypso.localhost:3000/new )
  * Go through gutenboarding steps
  * Sandbox that new site
  * Land on editor page, save url
  * Ensure the dev version of the FSE plugin is loading
  * On the editor page, Click launch button
  * Now you should see the existing domain and plan upsells (they should be hidden on the anchor path only)

fixes 391-gh-Automattic/dotcom-manage
